### ISSUE TITLE
AutoComply not automatically complying to my wishes

### ISSUE THREAD

**Issue by developocalypse** [JKorf/Binance.Net#379]
This fails because quantity isn't quite as high as the minimum BTC trade size (0.0001).
```
var price = client.GetPrice("XLMBTC").Data.Price;//0.00000870M


var quantity = 0.0001M / price;//11.494252873563218390804597701M

var r = client.PlaceOrder("XLMBTC", OrderSide.Buy, OrderType.Limit, quantity, null, null, price, TimeInForce.GoodTillCancel);

```
I want AutoComply to be changed so that it shunts the quantity up a tiny bit so the placement works. If that's impossible, what's the minimum I need to do to my code to make it work, other than by hard-coding a higher min BTC amount or other kludge?

**Comment by JKorf** [comment]
I've added checking for the MinNotional filter in the trade rules in the latest version. This should round the quantity up to check for the minimal BTC quantity in your case.

**Comment by developocalypse** [comment]
Thanks

**Comment by developocalypse** [comment]
The second buy fails
```
var btcAmount = 0.0001M;

var badPrice = 0.00005919M;
var goodPrice = 0.00005879M;

//Success
var goodResult = client.PlaceOrder("ONTBTC", OrderSide.Buy, OrderType.Limit, btcAmount / goodPrice, null, null, goodPrice, TimeInForce.GoodTillCancel);

//-1013: Filter failure: MIN_NOTIONAL
var badResult = client.PlaceOrder("ONTBTC", OrderSide.Buy, OrderType.Limit, btcAmount / badPrice, null, null, badPrice, TimeInForce.GoodTillCancel);
```

**Comment by JKorf** [comment]
Can you paste the output in the logs when it fails? Seems to me like it should work.

**Comment by developocalypse** [comment]
2020/08/05 11:13:02:560 | Info | Quantity clamped from 1.6894745734076702145632708228 to 1.68 based on lot size filter
2020/08/05 11:13:02:560 | Info | price rounded from 0.00005919 to 0.0000591 based on price filter
2020/08/05 11:13:02:561 | Info | Quantity clamped from 1.68 to 1.69 based on min notional filter
2020/08/05 11:13:02:562 | Debug | Creating request for https://api.binance.com/api/v3/order
2020/08/05 11:13:02:562 | Debug | Sending POST signed request to https://api.binance.com/api/v3/order with request body price=0.0000591&quantity=1.69&recvWindow=30000&side=BUY&symbol=ONTBTC&timeInForce=GTC&timestamp=1596622382547&type=LIMIT
2020/08/05 11:13:02:837 | Debug | Error received: {"code":-1013,"msg":"Filter failure: MIN_NOTIONAL"}

**Comment by developocalypse** [comment]
Version="5.1.14"

**Comment by JKorf** [comment]
Its because the price was also rounded, the trade rules check didn't take that into account when checking min notional. Will be fixed in next version.

**Comment by JKorf** [comment]
This has been fixed in the new 6.0.0 beta version. You'll have to do some refactoring using the new 6.0.0 version though since I've combined/moved/restructured things.

**Comment by developocalypse** [comment]
Cool, thanks. I'll test it soon.
